### PR TITLE
DEPR: Series.put, Series.real, Series.imag, Index.dtype_str

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] xref #https://github.com/pandas-dev/pandas/issues/18262
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Removes:
`Series.put`
`Index.dtype_str`
`Series.real`
`Series.imag`

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -804,7 +812,11 @@ def real(self, v):
     def imag(self):
         """
         Return imag value of vector.
+
+        .. deprecated 0.25.0
         """
+        warnings.warn("`imag` has be deprecated and will be removed in a "
+                      "future verison", DeprecationWarning, stacklevel=2)
```
Comment (jreback) [inline pandas/core/series.py]: DeprecationWarning -> FutureWaring on all of these

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -604,6 +604,9 @@ Other deprecations
 - The :meth:`Series.ftype`, :meth:`Series.ftypes` and :meth:`DataFrame.ftypes` methods are deprecated and will be removed in a future version.
   Instead, use :meth:`Series.dtype` and :meth:`DataFrame.dtypes` (:issue:`26705`).
 - :meth:`Timedelta.resolution` is deprecated and replaced with :meth:`Timedelta.resolution_string`.  In a future version, :meth:`Timedelta.resolution` will be changed to behave like the standard library :attr:`timedelta.resolution` (:issue:`21344`)
+- :attr:`Index.dtype_str` is deprecated. (:issue:`18262`)
+- :attr:`Series.imag` and :attr:`Series.real` is deprecated. (:issue:`18262`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: is -> are

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm, we should use FutureWarning here

**Comment by jreback** [comment]
can you add to the list in #6581 as well

**Comment by mroeschke** [comment]
Changed to FutureWarnings and green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**